server/auth_server.go

```diff
@@ -487,14 +484,16 @@ func RefreshToken(json map[string]string) (model.Token, error) {
 		}
 
 		payload := make(map[string]interface{})
+		if provider.GetName() == "ldap" {
+			payload["external_id"] = token.ExternalAccountID
+		}
```

Comment:
@mrajashree If cattle does not need the etxernal_id then if token.AccessToken is empty, we need not set token.JwtToken. I think that should work and help us avoid referencing provider with name in auth_server layer.